src/icalendar/cal.py

```diff
@@ -333,7 +333,7 @@ def from_ical(cls, st, multiple=False):
                 component = stack[-1] if stack else None
                 if not component or not component.ignore_exceptions:
                     raise
-                component.errors.append((None, str(e)))
+                component.errors.append((None, repr(e)))
```

Comment:
This is wrong. You can add e.message instead. You'll need to modify it at https://github.com/collective/icalendar/blob/master/src/icalendar/cal.py#L385 as well.